tardis/visualization/plot_util.py

```diff
@@ -79,3 +89,326 @@ def to_rgb255_string(color_tuple):
+        else:
+            full_species_list = []
+            species_mapped = {}
+            for species in species_list:
+                # check if a hyphen is present. If it is, then it indicates a
+                # range of ions. Add each ion in that range to the list as a new entry
+                if "-" in species:
+                    # split the string on spaces. First thing in the list is then the element
+                    parts = species.split(" ")
+                    element = parts[0]
+                    ion_range = parts[-1]
+                    # Next thing is the ion range
+                    # convert the requested ions into numerals
+                    range_parts = [
+                        part.strip() for part in ion_range.split("-")
+                    ]
+                    first_ion_numeral = roman_to_int(range_parts[0])
+                    second_ion_numeral = roman_to_int(range_parts[-1])
+                    # add each ion between the two requested into the species list
+                    for ion_number in np.arange(
+                        first_ion_numeral, second_ion_numeral + 1
+                    ):
+                        full_species_list.append(
+                            f"{element} {int_to_roman(ion_number)}"
+                        )
+                else:
+                    # Otherwise it's either an element or ion so just add to the list
+                    full_species_list.append(species)
+
+            # full_species_list is now a list containing each individual species requested
+            # e.g. it parses species_list = [Si I - V] into species_list = [Si I, Si II, Si III, Si IV, Si V]
+            requested_species_ids = []
+            keep_colour = []
+
+            # go through each of the requested species. Check whether it is
+            # an element or ion (ions have spaces). If it is an element,
+            # add all possible ions to the ions list. Otherwise just add
+            # the requested ion
+            for species in full_species_list:
+                if " " in species:
+                    species_id = (
+                        species_string_to_tuple(species)[0],
+                        species_string_to_tuple(species)[1],
+                    )
+                    requested_species_ids.append([species_id])
+                    species_mapped[species_id] = [species_id]
+                else:
+                    atomic_number = element_symbol2atomic_number(species)
+                    species_ids = [
+                        (atomic_number, ion_number)
+                        for ion_number in np.arange(atomic_number)
+                    ]
+                    requested_species_ids.append(species_ids)
+                    species_mapped[(atomic_number, 0)] = species_ids
+                    # add the atomic number to a list so you know that this element should
+                    # have all species in the same colour, i.e. it was requested like
```

Comment:
What does colour have to do with this?

Reply:
all ions of one element will have the same colour.

Reply:
Okay